libs/cypress-indexeddb/src/lib/cypress-indexeddb.ts

```diff
@@ -0,0 +1,167 @@
+        if (db) {
+          return Promise.resolve(db);
+        }
+        const newVersion = versionConfiguredByUser || 2;
+        const request = window.indexedDB.open(databaseName, newVersion);
+        return new Promise<IDBDatabase>((resolve, reject) => {
+          request.onerror = reject;
+          request.onupgradeneeded = (event: any) => {
+            const db = event.target?.result;
+            DATABASES.set(databaseName, db);
+            resolve(db);
+          };
+        });
+      });
+    }
+  );
+
+  Cypress.Commands.add(
+    `createObjectStore`,
+    { prevSubject: true },
+    (database: IDBDatabase, storeName: string) => {
+      if (database?.constructor?.name !== 'IDBDatabase') {
+        throw new Error(
+          `You tried to use the 'getObjectStore' method without calling 'openIndexedDb' first`
+        );
+      }
+      return new Promise((resolve, reject) => {
+        try {
+          let store: IDBObjectStore;
+          if (database.objectStoreNames.contains(storeName)) {
+            store = database.transaction(storeName, 'readwrite').objectStore(storeName);
+          } else {
+            store = database.createObjectStore(storeName);
+          }
+          new Promise((resolve2, reject2) => {
+            store.transaction.oncomplete = resolve2;
+            store.transaction.onerror = reject2;
+            store.transaction.onabort = reject2;
+          }).then(() => {
+            resolve(store);
+          });
+        } catch (e) {
+          reject(e);
+        }
+      });
+    }
+  );
+
+  Cypress.Commands.overwrite('as', (originalAs, subject: any | IDBObjectStore, alias) => {
+    if (subject?.constructor?.name === 'IDBObjectStore') {
+      STORES.set(alias, subject);
+      return subject;
+    } else if (subject?.constructor?.name === 'IDBDatabase') {
+      DATABASES.set(alias, subject);
+      return subject;
+    } else {
+      return originalAs(alias);
+    }
+  });
+
+  Cypress.Commands.add('getDatabase', (alias: string) => {
+    const withoutAtSign = alias.substr(1);
+    if (DATABASES.has(withoutAtSign)) {
+      return Promise.resolve(DATABASES.get(withoutAtSign));
+    } else {
+      throw new Error(`could not find database with alias ${alias}`);
+    }
+  });
+
+  Cypress.Commands.add('getStore', (alias: string) => {
+    const withoutAtSign = alias.substr(1);
+    if (STORES.has(withoutAtSign)) {
+      return Promise.resolve(STORES.get(withoutAtSign));
+    } else {
+      throw new Error(`could not find store with alias ${alias}`);
+    }
+  });
+
+  Cypress.Commands.add(
+    `storeItem`,
+    { prevSubject: true },
+    (store: IDBObjectStore, key: string, value: unknown) => {
+      if (store?.constructor?.name !== 'IDBObjectStore') {
+        throw new Error(
+          `You tried to use the 'storeItem' method without calling 'getObjectStore' first`
+        );
+      }
+      try {
+        const openStore: IDBObjectStore = store.transaction.db
```

Comment:
we could reuse the `isIDBObjectStore` I've proposed above here. 

```suggestion
    (store: IDBObjectStore, key: string, value: unknown) => {
      if (!isIDBObjectStore(store)) {
        throw new Error(
          `You tried to use the 'storeItem' method without calling 'getObjectStore' first`
        );
      }
      try {
        const openStore: IDBObjectStore = store.transaction.db
```